core/commons/che-core-commons-gwt/src/main/java/org/eclipse/che/ide/util/Arrays.java

```diff
@@ -0,0 +1,132 @@
+/*******************************************************************************
+ * Copyright (c) 2012-2016 Codenvy, S.A.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Codenvy, S.A. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.che.ide.util;
+
+import com.google.common.annotations.Beta;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.lang.System.arraycopy;
+import static java.util.Arrays.copyOf;
+
+/**
+ * Utility methods to operate with arrays.
+ *
+ * @author Vlad Zhukovskiy
+ * @since 4.2.0
+ */
+@Beta
+public class Arrays {
+
```

Comment:
I work a lot with arrays on the client side, so to reduce duplicated code I created utility class for that. This code is fully compatible with GWT, but we can move it to another common module if need and annotate with `com.google.common.annotations.GwtCompatible` annotation.